AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Subscribe/Support/Model+Sort.swift

```diff
@@ -0,0 +1,156 @@
+                return false
+            }
+            return ModelValueCompare(value1Optional: value1Optional,
+                                     value2Optional: value2Optional)
+                .sortComparator(sortOrder: sortOrder)
+        case .int, .timestamp:
+            guard let value1Optional = value1 as? Int?, let value2Optional = value2 as? Int? else {
+                return false
+            }
+            return ModelValueCompare(value1Optional: value1Optional,
+                                     value2Optional: value2Optional)
+                .sortComparator(sortOrder: sortOrder)
+
+        case .double:
+            guard let value1Optional = value1 as? Double?, let value2Optional = value2 as? Double? else {
+                return false
+            }
+            return ModelValueCompare(value1Optional: value1Optional,
+                                     value2Optional: value2Optional)
+                .sortComparator(sortOrder: sortOrder)
+
+        case .date:
+            guard let value1Optional = value1 as? Temporal.Date?, let value2Optional = value2 as? Temporal.Date? else {
+                return false
+            }
+            return ModelValueCompare(value1Optional: value1Optional,
+                                     value2Optional: value2Optional)
+                .sortComparator(sortOrder: sortOrder)
+        case .dateTime:
+            guard let value1Optional = value1 as? Temporal.DateTime?,
+                  let value2Optional = value2 as? Temporal.DateTime? else {
+                return false
+            }
+            return ModelValueCompare(value1Optional: value1Optional,
+                                     value2Optional: value2Optional)
+                .sortComparator(sortOrder: sortOrder)
+
+        case .time:
+            guard let value1Optional = value1 as? Temporal.Time?, let value2Optional = value2 as? Temporal.Time? else {
+                return false
+            }
+            return ModelValueCompare(value1Optional: value1Optional,
+                                     value2Optional: value2Optional)
+                .sortComparator(sortOrder: sortOrder)
+        case .bool:
+            guard let value1Optional = value1 as? Bool?, let value2Optional = value2 as? Bool? else {
+                return false
+            }
+            if value1Optional == nil && value2Optional == nil {
+                return sortOrder == .ascending
+            } else if value1Optional == nil, value2Optional != nil {
+                return sortOrder == .ascending
+            } else if value1Optional != nil, value2Optional == nil {
+                return sortOrder == .descending
+            } else if let value1 = value1Optional, let value2 = value2Optional {
+                return sortOrder == .ascending ?
+                    value1.intValue < value2.intValue : value1.intValue > value2.intValue
+            }
```

Comment:
Nested switch statements look weird but I'm wondering if using a switch statement with tuples would make this more readable and easier to maintain

Reply:
updated to reuse `ModelValueCompare`